src/openloco/windows/options.cpp

```diff
@@ -1,12 +1,2053 @@
+
+    }
+
+    namespace display
+    {
+        static const gfx::ui_size_t _window_size = { 366, 144 };
+
+        namespace widx
+        {
+            enum
+            {
+                display_resolution = 10,
+                display_resolution_btn,
+                landscape_smoothing,
+                gridlines_on_landscape,
+                vehicles_min_scale,
+                vehicles_min_scale_btn,
+                station_names_min_scale,
+                station_names_min_scale_btn,
+                construction_marker,
+                construction_marker_btn,
+            };
+        }
+
+        static widget_t _widgets[] = {
+            common_options_widgets(_window_size, string_ids::options_title_display),
+            make_widget({ 183, 49 }, { 173, 12 }, widget_type::wt_18, 1, string_ids::display_resolution_label_format),
+            make_widget({ 344, 50 }, { 11, 10 }, widget_type::wt_11, 1, string_ids::dropdown),
+            make_widget({ 10, 65 }, { 346, 12 }, widget_type::checkbox, 1, string_ids::landscape_smoothing, string_ids::landscape_smoothing_tip),
+            make_widget({ 10, 80 }, { 346, 12 }, widget_type::checkbox, 1, string_ids::gridlines_on_landscape, string_ids::gridlines_on_landscape_tip),
+            make_widget({ 183, 94 }, { 173, 12 }, widget_type::wt_18, 1, string_ids::empty, string_ids::vehicles_min_scale_tip),
+            make_widget({ 344, 95 }, { 11, 10 }, widget_type::wt_11, 1, string_ids::dropdown, string_ids::vehicles_min_scale_tip),
+            make_widget({ 183, 109 }, { 173, 12 }, widget_type::wt_18, 1, string_ids::empty, string_ids::station_names_min_scale_tip),
+            make_widget({ 344, 110 }, { 11, 10 }, widget_type::wt_11, 1, string_ids::dropdown, string_ids::station_names_min_scale_tip),
+            make_widget({ 183, 124 }, { 173, 12 }, widget_type::wt_18, 1, string_ids::empty),
+            make_widget({ 344, 125 }, { 11, 10 }, widget_type::wt_11, 1, string_ids::dropdown),
+            widget_end(),
+        };
+
+        static window_event_list _events;
+
+        // 0x004BFB8C
+        static void on_mouse_up(window* w, widget_index wi)
+        {
+            switch (wi)
+            {
+                case common::widx::close_button:
+                    windowmgr::close(w);
+                    return;
+
+                case common::widx::tab_display:
+                case common::widx::tab_sound:
+                case common::widx::tab_music:
+                case common::widx::tab_regional:
+                case common::widx::tab_controls:
+                case common::widx::tab_miscellaneous:
+                    options::tab_on_mouse_up(w, wi);
+                    return;
+
+                case widx::landscape_smoothing:
+                {
+                    auto& cfg = openloco::config::get();
+                    // TODO: is there a better way to toggle a flag?
```

Comment:
@AaronVanGeffen Do you have any idea on resolving this comment?

Reply:
`^=` ?

Reply:
Yeah: `cfg.flags ^= config::flags::landscape_smoothing;`